library/firewall_lib.py

```diff
@@ -178,7 +178,39 @@
 except ImportError:
     HAS_FIREWALLD = False
 
-
+def handle_interface(zone,item,fw_zone,fw_settings,fw,fw_offline):
+    if fw_offline:
+        iface_zone_objs = []
+        for zone in fw.config.get_zones():
+            old_zone_obj = fw.config.get_zone(zone)
+            if item in old_zone_obj.interfaces:
+                iface_zone_objs.append(old_zone_ob)
+        if len(iface_zone_objs) > 1:
+            module.fail_json(
+                msg="ERROR: interface {0} is in {1} zone XML file, can only be in one".format(
+                    interface, len(iface_zone_objs)
+                )
+            )
+        old_zone_obj = iface_zone_obj[0]
+        if old_zone_obj.name != zone:
+            old_zone_settings = FirewallClientZoneSettings(
+                fw.config.get_zone_config(old_zone_obj)
+            )
+            old_zone_settings.removeInterface(item)
+            fw.config.set_zone_config(
+                old_zone_obj, old_zone_settings.settings
+            )
+            fw_settings.addInterface(item)
+            fw.config.set_zone_config(fw_zone, fw_settings.settings)
+    else:
+        old_zone_name = fw.config().getZoneOfInterface(item)
+        if old_zone_name != zone:
+            if old_zone_name:
+                old_zone_obj = fw.config().getZoneByName(old_zone_name)
+                old_zone_settings = old_zone_obj.getSettings()
+                old_zone_settings.removeInterface(item)
+                fw_zone.update(fw_settings)
```

Comment:
This `update()` looks redundant. This file calls `update()` once at the end already.